Fixes #219

sanguinariojoe opened the pull request:
As promised, I am coming with a verification code. Some notes:

- In contrast to your code @RyanDavies19 , this is running the OpenFAST tests (just the moordyn ones), and then it is automagically collecting and running them with MoorDynC.
- The tolerances cannot be so strict. Instead of rtol=2 and atol=1.9, I had to set rtol=1.5 and atol=1.5
- I am directly importing the pass_fail.py from openfast instead of vendoring it.
- It is running right now on https://github.com/core-marine-dev/MoorDyn/actions/runs/9596912401/job/26464823608. I hope it will finish just fine
- I tried that with the dev branch of openfast... Oh boy, our lines failure stuff seems to be very broken...

RyanDavies19 commented:
@sanguinariojoe thanks for putting this together and for the rewrite of the test approach, this looks really solid. I will work on getting more regression test cases uploaded that really test our capabilities, and I will make sure that the options sections are compatible with both versions. For the line failures in MD-C, that is something I need to fix. I just rewrote and fixed the MD-F ones but didn't have the time to do MD-C yet, so don't worry about that one for now.

sanguinariojoe commented:
I'll do that, sure thing! But it has to wait until next week. Tomorrow we
celebrate midsummer! Partyyyyyyyyy

On Thu, 20 Jun 2024, 18:11 RyanDavies19, ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In tests/.mdf_verification/verify.py
> <https://github.com/FloatingArrayDesign/MoorDyn/pull/224#discussion_r1647825242>
> :
>
> > +        rod_state = moordyn.GetRodState(rod)
> +        state += np.array(rod_state).flatten().tolist()
> +        names += [f"rod_{rod_id}",] * len(rod_state)
> +    n_points = moordyn.GetNumberPoints(system)
> +    for i in range(n_points):
> +        point = moordyn.GetPoint(system, i + 1)
> +        if moordyn.GetPointType(point) not in (moordyn.POINT_TYPE_COUPLED, ):
> +            continue
> +        point_id = moordyn.GetPointID(point)
> +        point_state = moordyn.GetPointPos(point)
> +        state += point_state
> +        names += [f"point_{point_id}",] * len(point_state)
> +    return state, names
> +
> +
> +def read_outs(fpath, skiplines=2):
>
> Add a check here for *** akin to:
> https://github.com/RyanDavies19/MoorDynC_ryan/blob/c572051566b119b401c623fe6c3f42d8b085dcc3/verification/verify.py#L51
>
> OpenFAST/MD-F will sometimes have an overflow error for numbers very close
> to 0 (e-100) where it prints *** instead of the value
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FloatingArrayDesign/MoorDyn/pull/224#pullrequestreview-2130853055>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMXKKGK7HITLKCPL6J26OLZIL5M7AVCNFSM6AAAAABJTZTF2SVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMZQHA2TGMBVGU>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sanguinariojoe commented:
BTW, I am idiot and pointed the PR against master. Can you change that to
point to dev?

Otherwise I'll do that next week

On Thu, 20 Jun 2024, 18:06 RyanDavies19, ***@***.***> wrote:

> @sanguinariojoe <https://github.com/sanguinariojoe> thanks for putting
> this together and for the rewrite of the test approach, this looks really
> solid. I will work on getting more regression test cases uploaded that
> really test our capabilities, and I will make sure that the options
> sections are compatible with both versions. For the line failures in MD-C,
> that is something I need to fix. I just rewrote and fixed the MD-F ones but
> didn't have the time to do MD-C yet, so don't worry about that one for now.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/FloatingArrayDesign/MoorDyn/pull/224#issuecomment-2181056962>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAMXKKDB4ZST64USB5JG4FLZIL4YTAVCNFSM6AAAAABJTZTF2SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOBRGA2TMOJWGI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

RyanDavies19 commented:
> BTW, I am idiot and pointed the PR against master. Can you change that to point to dev? Otherwise I'll do that next week
> […](#)
> On Thu, 20 Jun 2024, 18:06 RyanDavies19, ***@***.***> wrote: @sanguinariojoe <https://github.com/sanguinariojoe> thanks for putting this together and for the rewrite of the test approach, this looks really solid. I will work on getting more regression test cases uploaded that really test our capabilities, and I will make sure that the options sections are compatible with both versions. For the line failures in MD-C, that is something I need to fix. I just rewrote and fixed the MD-F ones but didn't have the time to do MD-C yet, so don't worry about that one for now. — Reply to this email directly, view it on GitHub <[#224 (comment)](https://github.com/FloatingArrayDesign/MoorDyn/pull/224#issuecomment-2181056962)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAMXKKDB4ZST64USB5JG4FLZIL4YTAVCNFSM6AAAAABJTZTF2SVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOBRGA2TMOJWGI> . You are receiving this because you were mentioned.Message ID: ***@***.***>

Done

sanguinariojoe commented on tests/.mdf_verification/verify.py:
```diff
@@ -0,0 +1,329 @@
+    return state, names
+
+
+def read_outs(fpath, skiplines=2):
```
Done in e51f22703bf875d7a22fcc4ef77939135e2e2ade
